rav1e_js/www/src/App.tsx

```diff
@@ -7,88 +7,48 @@
 // Media Patent License 1.0 was not distributed with this source code in the
 // PATENTS file, you can obtain it at www.aomedia.org/license/patent.
 
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 
-import { EncoderConfigDetails, HelpAndAbout } from "./components";
-import { Encoder, EncoderConfig, Packet } from "rav1e";
-import { EncoderStatus } from "./utils";
-
-// Configuration for encoder
-const conf = new EncoderConfig()
-	.setDim(64, 96)
-	.setRdoLookaheadFrame(1);
-
-// JSON-string representing configuration
-const configStr = conf.toJSON();
-
-// Encoder
-const enc = new Encoder(conf);
-
-// Frame to send to encoder
-const f = enc.newFrame();
+import { ChromaSampling, EncoderConfig, Frame, VideoEncoder } from "rav1e";
 
 export default function App() {
-	const [encoderStatus, setEncoderStatus] = useState(EncoderStatus.Initialized);
-	const [flushing, setFlushing] = useState(false);
-	const [frameQ_length, setFrameQ_length] = useState(0);
-	const [packets, setPackets] = useState(Array<Packet>())
-
-	const sendFrame = () => {
-		enc.sendFrame(f);
-		setFrameQ_length(frameQ_length + 1);
-		setEncoderStatus(EncoderStatus.Received)
-	};
-
-	const flush = () => {
-		enc.flush();
-		setFlushing(true);
-		setEncoderStatus(EncoderStatus.Flushed);
-	}
-
-	const receivePacket = () => {
-		try {
-			const p = enc.receivePacket();
-			setEncoderStatus(EncoderStatus.Encoded);
-
-			// append new packet
-			const newPackets = Array.from(packets);
-			newPackets.push(p);
-			setPackets(newPackets);
-
-			// needed to receive the "limit reached" status
-			if (flushing && (frameQ_length - newPackets.length) === 0) {
-				receivePacket();
-			}
-		} catch (e) {
-			if (e === "encoded") {
-				receivePacket();
-			} else if (e === "need more data") {
-				setEncoderStatus(EncoderStatus.NeedMoreData);
-			} else if (e === "limit reached") {
-				setEncoderStatus(EncoderStatus.LimitReached);
-			} else {
-				setEncoderStatus(e);
-				console.error(e);
+	// useEffect, because then it runs after the browser rendered the display content
+	useEffect(() => {
+		const video = document.getElementById("video") as HTMLVideoElement;
+		console.log(video.videoWidth, video.videoHeight);
+
+		video.onended = (e) => {
+			enc.flush();
+			console.log("flushed")
+
+			// encode all frames
+			while (true) {
+				try {
+					const p = enc.receivePacket();
```

Comment:
when I was trying I couldn't see/catch any of these, so please document first how to do before saying if it is my issue or the app as a whole

Reply:
Okay :+1: